src/main/java/io/github/christophermanahan/carnitas/HTTPServer.java

```diff
@@ -25,8 +25,9 @@ private void connect() {
     connection = listener.listen();
   }
 
-  private void echo() {
+  private void serve() {
     connection.receive()
+      .map(HTTPResponse::new)
```

Comment:
Looks cool, but we might be better off doing something like this until we have more information.

```suggestion
      .map(request -> new HTTPResponse())
```